packages/opentelemetry-core/src/context/propagation/B3Propagator.ts

```diff
@@ -38,6 +48,77 @@ function isValidSpanId(spanId: string): boolean {
   return VALID_SPANID_REGEX.test(spanId) && !INVALID_ID_REGEX.test(spanId);
 }
 
+function isValidParentSpanID(spanId: string | undefined): boolean {
+  return spanId === undefined || isValidSpanId(spanId);
+}
+
+function isValidSampledValue(sampled: TraceFlags | undefined): boolean {
+  return sampled === TraceFlags.SAMPLED || sampled === TraceFlags.NONE;
+}
+
+function parseHeader(header: unknown) {
+  return Array.isArray(header) ? header[0] : header;
+}
+
+function getHeaderValue(carrier: unknown, getter: GetterFunction, key: string) {
+  const header = getter(carrier, key);
+  return parseHeader(header);
+}
+
+function getTraceId(carrier: unknown, getter: GetterFunction): string {
+  const traceId = getHeaderValue(carrier, getter, X_B3_TRACE_ID);
+  if (typeof traceId === 'string') {
+    return traceId.padStart(32, '0');
+  }
+  return '';
+}
+
+function getSpanId(carrier: unknown, getter: GetterFunction): string {
+  const spanId = getHeaderValue(carrier, getter, X_B3_SPAN_ID);
+  if (typeof spanId === 'string') {
+    return spanId;
+  }
+  return '';
+}
+
+function getParentSpanId(
+  carrier: unknown,
+  getter: GetterFunction
+): string | undefined {
+  const spanId = getHeaderValue(carrier, getter, X_B3_PARENT_SPAN_ID);
+  if (typeof spanId === 'string') {
+    return spanId;
+  }
+  return;
+}
+
+function getDebug(
+  carrier: unknown,
+  getter: GetterFunction
+): string | undefined {
+  const debug = getHeaderValue(carrier, getter, X_B3_FLAGS);
+  return debug === '1' ? '1' : undefined;
+}
+
+function getTraceFlags(
+  carrier: unknown,
+  getter: GetterFunction
+): TraceFlags | undefined {
+  const traceFlags = getHeaderValue(carrier, getter, X_B3_SAMPLED);
+  const debug = getDebug(carrier, getter);
+  if (debug === '1' || VALID_SAMPLED_VALUES.has(traceFlags)) {
+    return TraceFlags.SAMPLED;
+  }
+  if (VALID_UNSAMPLED_VALUES.has(traceFlags)) {
+    return TraceFlags.NONE;
+  }
+  if (traceFlags === undefined) {
```

Comment:
I would prefer to unify this and above if block.

Reply:
cool, will combine now